src/borg/archiver/prune_cmd.py

```diff
@@ -104,24 +119,30 @@ def prune_split(archives, rule, n, kept_because=None):
         period = period_func(a)
         if period != last:
             last = period
-            if a.id not in kept_because:
+            if a.id not in kept_because and (interval is None or a.ts >= base_timestamp - interval):
```

Comment:
Since the archives are sorted, once `a.ts` becomes too old, we should be able to abort the loop.  As the first line inside the `for a` loop, we could have:
```
if interval is not None and a.ts < base_timestamp - interval:
    break
```
I think this makes the logic more clear as well.  (But please carefully check my thinking as I've only skimmed the code.)

A separate point is that we could precompute `base_timestamp - inverval` outside of the loop.  Moreover, if `interval` is None, can we just set the precomputed value to the earliest timestamp, simplifying the condition to
```
if a.ts < interval_start:
    break
```
?

Reply:
Good suggestions, I'll implement all this.